pkg/http/api_directmessaging.go

```diff
@@ -191,12 +191,12 @@ func (a *api) onDirectMessage(w http.ResponseWriter, r *http.Request) {
 		} else if resStatus.Code < 200 || resStatus.Code > 399 {
 			msg, _ := rResp.RawDataFull()
 			// Returning a `codeError` here will cause Resiliency to retry the request (if retries are enabled), but if the request continues to fail, the response is sent to the user with whatever status code the app returned.
-			return rResp, codeError{
+			return rResp, resiliency.NewCodeError(resStatus.Code, codeError{
```

Comment:
We need this change for gRPC too?

Reply:
Sorry for the late reply, it takes me some time to look into the code.

Added for grpc_proxy mode, along with the e2e tests.

For http directmessaging, if the application uses the grpc protocol, when the application returns a grpc codes error, it is parsed into the rResp and does not treat it as an error(so would not retry). see: https://github.com/dapr/dapr/blob/master/pkg/channel/grpc/grpc_channel.go#L126-L133
https://github.com/dapr/dapr/blob/master/pkg/http/api_directmessaging.go#L173-L190
So currently not added, If we want to add it for this case, I think it is worth opening a new issue.

Reply:
We need to cover the case of gRPC -> gRPC and gRPC -> HTTP too. These are legacy methods but still supported.

Reply:
We don't support gRPC to HTTP.